### PR TITLE
Add application menu

### LINKED ISSUE #549: Open preferences hotkey does not work
**macOS version:**    darwin x64 17.7.0
**Kap version:**      3.0.0-beta.1

#### Steps to reproduce
1. Click Kap in the menubar and give it focus
2. Use the hotkey `CMD + ,` to Open Preferences

#### Current behavior

Nothing happens

#### Expected behavior

Preferences page opens

#### Workaround

Click `...` in the overlay and then select "Preferences..."

### LINKED ISSUE #552: Show Kap in the dock when it has windows open
This basically reintroduces behaviour we had in Kap 1.0, when a window (besides the cropper) is open Kap is shown in the dock, allowing you to tab to it.

### PR DESCRIPTION
**Pull request by karaggeorge**
Fixes #549 
Fixes #552 

Adds application menu and dock along with window titles. Enables all the common shortcuts (Copy, Cut, Paste etc)

Had the change the exports window implementation. Used to always be open and just shown/hidden. Switched to actually opening/closing it so I can show/hide the dock when there are no windows. Added a 'exports-ready' event to show the window when the rendering has finished

kapture 2019-01-29 at 10 50 19

### CODE REVIEW

**Comment by karaggeorge** [comment]
Questions/Discussions from Slack: 

- Under the Window menu the editors show up with the file name. For recordings just made that's a long temp file name that gives the user little to no information. We could replace that with something like `tmp-recording-1.mp4` or just show 'New Recording` or some other generic title. If you've done more than one recording you probably won't use that menu or `tmp-recording-{num}.mp4` to figure out which one is which.

**Comment by skllcrn** [comment]
Maybe we can use the timestamp of the recording for the temporary name @karaggeorge?

**Comment by karaggeorge** [comment]
@skllcrn `New Recording - 09:44:13am`?

**Comment by skllcrn** [comment]
I'm not sure, our current convention is e.g. `2018-11-29 at 17.10.36`. Thoughts @sindresorhus?

File: main/menus.js [hunk 1]
```diff
@@ -36,28 +36,105 @@ Workaround:           A workaround for the issue if you've found on. (this will
...
+    }, filePaths => {
+      if (filePaths) {
+        for (const file of filePaths) {
+          openEditorWindow(file);
+        }
+      }
+    });
+  }
+}, {
+  label: 'Export History…',
```
Comment (sindresorhus) [inline main/menus.js]: Remove the `...`

**Comment by sindresorhus** [comment]
I would use only time unless the recordings are from the day before, then I would include date. The common case will then only see time, which looks better.

**Comment by sindresorhus** [comment]
The app menu shows when the cropper opens, it should not show for the cropper.

File: main/menus.js [hunk 2]
```diff
@@ -76,38 +153,88 @@ const cogMenuTemplate = [
...
+  role: 'window',
+  submenu: [{
+    label: 'Minimize',
+    accelerator: 'CmdOrCtrl+M',
+    role: 'minimize'
+  }, {
+    label: 'Close',
+    accelerator: 'CmdOrCtrl+W',
+    role: 'close'
+  }]
```
[the remark this answers is not in the document]
Comment (karaggeorge) [reply main/menus.js]: Is there a role for the zoom and bring all to front? Right now I got the minimize, export history and the list of windows

screen shot 2019-01-31 at 9 44 14 am

**Comment by karaggeorge** [comment]
@sindresorhus updated based on your feedback. 

For the time stamp, when you create the recording it's always going to be the current day and the window is created at that point, so if even if it's left open for a while, the title will stay the same. Not sure how/when we would show the date

cc @skllcrn

File: main/menus.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sindresorhus) [reply main/menus.js]: You can see all the roles here: https://electronjs.org/docs/api/menu-item

You want `zoom` and `front`.

It should be in this order:

```
Minimize
Zoom
---------
Export History
---------
Bring All to Front
---------
Kapture 212321321312
```

[the remark this answers is not in the document]
Comment (karaggeorge) [reply main/menus.js]: That's the way it is in the PR right now

**Comment by sindresorhus** [comment]
> For the time stamp, when you create the recording it's always going to be the current day and the window is created at that point, so if even if it's left open for a while, the title will stay the same. Not sure how/when we would show the date

Ah, I didn't think of that. I guess we need date on all of them then.

File: main/menus.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sindresorhus) [reply main/menus.js]: Yeah, I just noticed. I read the comments before looking at the commits.

**Comment by karaggeorge** [comment]
@sindresorhus Added the date to all of them and resolved conflicts 👍

**Comment by sindresorhus** [comment]
Looks good 👌

**Pull request merged**